web3/_utils/contracts.py

```diff
@@ -124,11 +114,11 @@ def find_matching_event_abi(
     argument_names: Optional[Sequence[str]] = None,
 ) -> ABIEvent:
     filters = [
-        functools.partial(filter_by_type, "event"),
+        functools.partial(filter_abi_by_type, "event"),
     ]
 
     if event_name is not None:
-        filters.append(functools.partial(filter_by_name, event_name))
+        filters.append(functools.partial(_filter_event_abi_by_name, event_name))
```

Comment:
Can we just call `filter_abi_by_name` here? It looks like `_filter_event_abi_by_name` is doing the same thing, but I might be mistaken